src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -150,5 +160,19 @@ static IEnumerable<Diagnostic> GetDynamicallyAccessedMembersDiagnostics (SingleV
 
 			return diagnosticContext.Diagnostics;
 		}
+
+		static void VerifyMemberOnlyApplyToTypesOrStrings (SymbolAnalysisContext context, ISymbol member)
+		{
+			if (member is IFieldSymbol field && field.GetDynamicallyAccessedMemberTypes () != DynamicallyAccessedMemberTypes.None && !field.Type.IsTypeInterestingForDataflow ())
```

Comment:
Maybe we should use `TryGetAttribute` here so that if you explicitly put `DAMT.None` on an unsupported type, it still produces a warning. What does the linker do in that case?

Reply:
Linker does not warn on this case, should I proceed to warn even if this differs from linker's behavior?

Reply:
Then I would just follow the linker behavior, and maybe open an issue as a future improvement.

Reply:
Open https://github.com/dotnet/linker/issues/2603 to track this discussion